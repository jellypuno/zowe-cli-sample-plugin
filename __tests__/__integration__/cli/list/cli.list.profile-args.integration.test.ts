/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import * as fs from "fs";
import * as path from "path";
import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { IO } from "@zowe/imperative";
import { ITestPropertiesSchema } from "../../../__src__/environment/doc/ITestPropertiesSchema";

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;

const configJson = "zowe.config.json";
const configUserJson = "zowe.config.user.json";
const baseYaml = "my_base.yaml";
const sampleYaml = "my_sample.yaml";

describe("zowe-cli-sample list profile-args command", () => {
    // Create the unique test environment
    beforeEach(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            installPlugin: true,
            testName: "list_profile_args_command",
            skipProperties: true
        });
    });

    afterEach(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should list profile args from team config profile and other sources", () => {
        fs.copyFileSync(path.join(__dirname, "__resources__", configJson), path.join(TEST_ENVIRONMENT.workingDir, configJson));
        fs.copyFileSync(path.join(__dirname, "__resources__", configUserJson), path.join(TEST_ENVIRONMENT.workingDir, configUserJson));

        const response = runCliScript(__dirname + "/__scripts__/list_profile_args.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        const output = response.stdout.toString();
        expect(output).toMatch(/host:\s+new.host.com/);
        expect(output).toMatch(/port:\s+1337/);
        expect(output).toMatch(/user:\s+user1/);
        expect(output).toMatch(/password:\s+123456/);
        expect(output).toMatch(/usingTeamConfig:\s+true/);
        expect(output).toMatch(/sampleProfileName:\s+my_sample/);
        expect(output).toMatch(/baseProfileName:\s+my_base/);
        expect(output).toMatchSnapshot();
    });

    it("should list profile args from old school profile and other sources", () => {
        IO.mkdirp(path.join(TEST_ENVIRONMENT.workingDir, "profiles", "base"));
        fs.copyFileSync(path.join(__dirname, "__resources__", baseYaml), path.join(TEST_ENVIRONMENT.workingDir, "profiles", "base", baseYaml));
        IO.mkdirp(path.join(TEST_ENVIRONMENT.workingDir, "profiles", "sample"));
        fs.copyFileSync(path.join(__dirname, "__resources__", sampleYaml), path.join(TEST_ENVIRONMENT.workingDir, "profiles", "sample", sampleYaml));

        const response = runCliScript(__dirname + "/__scripts__/list_profile_args.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        const output = response.stdout.toString();
        expect(output).toMatch(/host:\s+new.host.com/);
        expect(output).toMatch(/port:\s+1337/);
        expect(output).toMatch(/user:\s+user1/);
        expect(output).toMatch(/password:\s+123456/);
        expect(output).toMatch(/usingTeamConfig:\s+false/);
        expect(output).toMatch(/sampleProfileName:\s+my_sample/);
        expect(output).toMatch(/baseProfileName:\s+my_base/);
        expect(output).toMatchSnapshot();
    });
});
