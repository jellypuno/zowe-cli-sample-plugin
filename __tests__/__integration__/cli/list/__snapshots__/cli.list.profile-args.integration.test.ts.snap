// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zowe-cli-sample list profile-args command should list profile args from old school profile and other sources 1`] = `
"Profile created successfully! Path:
C:\\\\Users\\\\tj891237\\\\Projects\\\\zowe\\\\zowe-cli-sample-plugin\\\\__tests__\\\\__results__\\\\data\\\\f4208b71-37b0-4cf2-8405-e8e23daecf72_list_profile_args_command\\\\profiles\\\\base\\\\my_base.yaml

host:               old.host.com
user:               admin
rejectUnauthorized: true

Review the created profile and edit if necessary using the profile update command.
Profile created successfully! Path:
C:\\\\Users\\\\tj891237\\\\Projects\\\\zowe\\\\zowe-cli-sample-plugin\\\\__tests__\\\\__results__\\\\data\\\\f4208b71-37b0-4cf2-8405-e8e23daecf72_list_profile_args_command\\\\profiles\\\\sample\\\\my_sample.yaml

port:               443
user:               user1
password:           123456
rejectUnauthorized: true
tshirtSize:         M

Review the created profile and edit if necessary using the profile update command.
arguments: 
  host:               new.host.com
  port:               1337
  user:               user1
  password:           123456
  rejectUnauthorized: true
  tshirtSize:         M
environment: 
  usingTeamConfig:   false
  sampleProfileName: my_sample
  baseProfileName:   my_base
"
`;
